waspc/src/Wasp/Generator/DbGenerator.hs

```diff
@@ -99,7 +110,49 @@ genPrismaSchema wasp = createTemplateFileDraft dstPath tmplSrcPath (Just templat
 
 genMigrationsDir :: Wasp -> Maybe FileDraft
 genMigrationsDir wasp =
-  (getMigrationsDir wasp) >>= \waspMigrationsDir ->
+  getMigrationsDir wasp >>= \waspMigrationsDir ->
     Just $ createCopyDirFileDraft (SP.castDir genProjectMigrationsDir) (SP.castDir waspMigrationsDir)
   where
     genProjectMigrationsDir = dbRootDirInProjectRootDir </> dbMigrationsDirInDbRootDir
+
+writeDbSchemaChecksumToFile :: Path' Abs (Dir ProjectRootDir) -> IO ()
+writeDbSchemaChecksumToFile genProjectRootDir = do
+  dbSchemaExists <- doesFileExist dbSchemaFp
+  when dbSchemaExists $ do
+    checksum <- calculateChecksumStringForFile dbSchemaFp
+    writeFile dbSchemaChecksumFp checksum
+  where
+    dbSchemaFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaFileInProjectRootDir
+    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaChecksumFileInProjectRootDir
+
+checkIfDbSchemaMatchesChecksum :: Path' Abs (Dir ProjectRootDir) -> ExceptT String IO ()
```

Comment:
I would make the name more elaborate, since it might not be clear what does the `Checksum` in the name stand for.
So, it could be `checkIfDbSchemaMatchesDbSchemaChecksumFile`. It is long, but I think it tells more correctly what it is about. It could even be smth like `checkIfDbSchemaWasUsedForLastMigration` -> maybe this is even better, since it hides implementation details and more clearly says what we want to use this function for. or smth in this direction, whatever is semantically most correct. I am not sure what makes most sense here, so you decide, I am just throwing ideas.

Reply:
Yep, I like the second one better myself too, thanks! Will update 👍🏻